src/utils/simplifier.js

```diff
@@ -9,22 +9,23 @@
 
 /* @flow */
 
+import type { BabelNodeSourceLocation } from "babel-types";
 import { FatalError } from "../errors.js";
 import { ValuesDomain } from "../domains/index.js";
 import invariant from "../invariant.js";
 import { ToBoolean } from "../methods/index.js";
 import { Realm } from "../realm.js";
 import { AbstractObjectValue, AbstractValue, BooleanValue, ConcreteValue, Value } from "../values/index.js";
 
-export default function simplifyAbstractValue(realm: Realm, value: AbstractValue): Value {
+export default function simplifyAbstractValue(realm: Realm, value: AbstractValue, isCondition: boolean = false): Value {
```

Comment:
I don't understand the intuition behind `isCondition`.

Reply:
x || true can simplify to just true if we don't care about the actual value of x, only its truthiness.